File: TransCoder_tokenized_test_set_functions/python/EULERIAN_PATH_UNDIRECTED_GRAPH.py
def findpath ( graph ) :
    n = len ( graph )
    numofadj = list ( )
    for i in range ( n ) :
        numofadj . append ( sum ( graph [ i ] ) )
    startpoint = 0
    numofodd = 0
    for i in range ( n - 1, - 1, - 1 ) :
        if ( numofadj [ i ] % 2 == 1 ) :
            numofodd += 1
            startpoint = i
    if ( numofodd > 2 ) :
        print ( "No Solution" )
        return
    stack = list ( )
    path = list ( )
    cur = startpoint
    while ( stack != [ ] or sum ( graph [ cur ] ) != 0 ) :
        if ( sum ( graph [ cur ] ) == 0 ) :
            path . append ( cur + 1 )
            cur = stack . pop ( - 1 )
        else :
            for i in range ( n ) :
                if graph [ cur ] [ i ] == 1 :
                    stack . append ( cur )
                    graph [ cur ] [ i ] = 0
                    graph [ i ] [ cur ] = 0
                    cur = i
                    break
    for ele in path :
        print ( ele, "-> ", end = '' )
    print ( cur + 1 )

File: TransCoder_tokenized_test_set_functions/java/EULERIAN_PATH_UNDIRECTED_GRAPH.java
static void findpath ( int [ ] [ ] graph, int n ) {
  Vector < Integer > numofadj = new Vector < > ( );
  for ( int i = 0;
  i < n;
  i ++ ) numofadj . add ( accumulate ( graph [ i ], 0 ) );
  int startPoint = 0, numofodd = 0;
  for ( int i = n - 1;
  i >= 0;
  i -- ) {
    if ( numofadj . elementAt ( i ) % 2 == 1 ) {
      numofodd ++;
      startPoint = i;
    }
  }
  if ( numofodd > 2 ) {
    System . out . println ( "No Solution" );
    return;
  }
  Stack < Integer > stack = new Stack < > ( );
  Vector < Integer > path = new Vector < > ( );
  int cur = startPoint;
  while ( ! stack . isEmpty ( ) || accumulate ( graph [ cur ], 0 ) != 0 ) {
    if ( accumulate ( graph [ cur ], 0 ) == 0 ) {
      path . add ( cur );
      cur = stack . pop ( );
    }
    else {
      for ( int i = 0;
      i < n;
      i ++ ) {
        if ( graph [ cur ] [ i ] == 1 ) {
          stack . add ( cur );
          graph [ cur ] [ i ] = 0;
          graph [ i ] [ cur ] = 0;
          cur = i;
          break;
        }
      }
    }
  }
  for ( int ele : path ) System . out . print ( ele + " -> " );
  System . out . println ( cur );
}